File: common/src/main/java/cc/turtl/chiselmon/config/category/RecorderConfig.java
package cc.turtl.chiselmon.config.category;

import cc.turtl.chiselmon.config.OptionFactory;
import dev.isxander.yacl3.api.ConfigCategory;
import dev.isxander.yacl3.config.v2.api.SerialEntry;
import net.minecraft.client.gui.screens.Screen;
import net.minecraft.network.chat.Component;

public class RecorderConfig {

    public static final boolean DEFAULT_ACTION_BAR = true;
    public static final boolean DEFAULT_DESPAWN_GLOW = false;

    @SerialEntry
    public boolean actionBar = DEFAULT_ACTION_BAR;
    @SerialEntry
    public boolean despawnGlow = DEFAULT_DESPAWN_GLOW;

    public ConfigCategory buildCategory(Screen parent) {
        return ConfigCategory.createBuilder()
                .name(Component.translatable("chiselmon.config.category.recorder"))
                .option(OptionFactory.toggleOnOff(
                        "chiselmon.config.recorder.action_bar",
                        DEFAULT_ACTION_BAR,
                        () -> actionBar,
                        v -> actionBar = v
                ))
                .option(OptionFactory.toggleOnOff(
                        "chiselmon.config.recorder.despawn_glow",
                        DEFAULT_DESPAWN_GLOW,
                        () -> despawnGlow,
                        v -> despawnGlow = v
                ))
                .build();
    }
}

File: common/src/main/kotlin/cc/turtl/chiselmon/client/config/category/RecorderConfig.kt
package cc.turtl.chiselmon.client.config.category

import cc.turtl.turtlshell.api.client.config.OptionFactory
import dev.isxander.yacl3.api.ConfigCategory
import dev.isxander.yacl3.config.v2.api.SerialEntry
import net.minecraft.network.chat.Component

class RecorderConfig {

    @SerialEntry
    var actionBar: Boolean = DEFAULT_ACTION_BAR

    @SerialEntry
    var despawnGlow: Boolean = DEFAULT_DESPAWN_GLOW

    fun buildCategory(): ConfigCategory = ConfigCategory.createBuilder()
        .name(Component.translatable("chiselmon.config.category.recorder"))
        .option(
            OptionFactory.toggleOnOff(
                "chiselmon.config.recorder.action_bar",
                DEFAULT_ACTION_BAR,
                { actionBar },
                { actionBar = it })
        )
        .option(
            OptionFactory.toggleOnOff(
                "chiselmon.config.recorder.despawn_glow",
                DEFAULT_DESPAWN_GLOW,
                { despawnGlow },
                { despawnGlow = it })
        )
        .build()

    companion object {
        const val DEFAULT_ACTION_BAR = true
        const val DEFAULT_DESPAWN_GLOW = false
    }
}